src/app/modules/main/chat_section/chat_content/messages/module.nim

```diff
@@ -585,11 +585,11 @@ method updateChatFetchMoreMessages*(self: Module) =
   if (self.controller.getChatDetails().hasMoreMessagesToRequest()):
     self.view.model().insertItemBasedOnClock(self.createFetchMoreMessagesItem())
 
-method getLinkPreviewData*(self: Module, link: string, uuid: string): string =
-  return self.controller.getLinkPreviewData(link, uuid)
+method getLinkPreviewData*(self: Module, link: string, uuid: string, whiteListedSites: string, whiteListedImgExtensions: string, unfurlImages: bool): string =
```

Comment:
I think the module or the controller could get the `whiteListedSites`, `whiteListedImgExtensions` and `unfurlImages`.
It feels a bit redundant for the backend to pass the UI those values, for the UI to pass it back again to the backend.
It would simplify those calls and a bit of the frontend too

Reply:
It was my first option, but unfortunately `whiteListedSites` is stored as QVariant in nim and internally it's a QVariantMap. I could not find any option to translate the QVariantMap into something nim would understand without extending nimqml. And I'm not sure it's worth doing it just for this. Since `whiteListedSites` made the most sense to me to be read directly from nim I gave up the idea completely.

But it's not a bad option IMO. Even if these variables would be read from nim we would need to bind on them in QML and request new previews on change.

Reply:
Expose as a JSON object? ;)

Reply:
Ah ok, no problem then, we can improve later if needed